### PR TITLE
DrivAerML: Lower LR (3e-4) at 4L/512d

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis
The new DrivAerML baseline (4.619% at 4L/512d) uses lr=5e-4. Lower learning rate was the breakthrough for TandemFoil (3e-4→2e-4 gave -14.7%). The wider 4L/512d model has more parameters and may benefit from slower, more stable optimization. Lower LR + 267 epochs (180-min budget) should provide smoother convergence without the instability seen at lr=5e-4.

## Instructions
Train on DrivAerML with 4L/512d architecture at lr=3e-4:

```bash
cd target/icml2026 && SENPAI_MAX_EPOCHS=9999 SENPAI_TIMEOUT_MINUTES=180 python train.py \
  --dataset drivaerml \
  --optimizer adamw \
  --lr 3e-4 \
  --cosine-t-max 30 \
  --no-use-ema \
  --enable-fourier \
  --model-layers 4 \
  --model-hidden-dim 512 \
  --model-heads 8 \
  --epochs 999 \
  --batch-size 1 \
  --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 \
  --max-eval-batches 200 \
  --agent ray \
  --wandb-name "ray/drivaerml-4L512d-lr3e4" \
  --wandb-group "drivaerml-4L512d-lr-sweep"
```

IMPORTANT: Must use `--batch-size 1 --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 --max-train-batches 394 --max-eval-batches 200`. Do not change these flags.

Report `val_primary/surface_rel_l2_pct` and note which epoch was best.

## Baseline
Current best: **4.619%** at epoch 256 (4L/512d/8H, T_max=30, lr=5e-4, 267 epochs, 180-min)
PR: #2691 | W&B: k8qtsxxz
External target: <3.71%

Reproduce: `cd target/icml2026 && SENPAI_MAX_EPOCHS=9999 SENPAI_TIMEOUT_MINUTES=180 python train.py --dataset drivaerml --optimizer adamw --lr 5e-4 --cosine-t-max 30 --no-use-ema --enable-fourier --model-layers 4 --model-hidden-dim 512 --model-heads 8 --epochs 999 --batch-size 1 --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 --max-train-batches 394 --max-eval-batches 200`

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT ray:

## Results

### DrivAerML 4L/512d lr=3e-4 (180-min budget)

| Metric | Baseline (lr=5e-4) | lr=3e-4 | Delta |
|--------|-------------------|---------|-------|
| val_primary/surface_rel_l2_pct (best) | **4.619%** (ep 256) | **4.937%** (ep 215) | +0.318 pp (+6.9% worse) |
| test_primary/surface_rel_l2_pct | — | 6.728% (final epoch) | — |
| Epochs completed | 267 | 267 | same |
| Best epoch | 256 | 215 | earlier peak |

**Top 5 epochs:** 215 (4.937%), 264 (4.968%), 253 (5.025%), 257 (5.046%), 213 (5.070%)

**Convergence profile:**
- Epoch 50: 11.96% → Epoch 100: 7.28% → Epoch 150: 6.72% → Epoch 200: 5.58% → Epoch 215: **4.94%** (best) → Epoch 267: 5.81%
- Same oscillation pattern as baseline (T_max=30 cosine cycling), but the troughs are shallower
- Best checkpoint came at epoch 215 vs baseline's 256 — the lower LR converges earlier but to a worse minimum

**Command used:**
```bash
cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 SENPAI_MAX_EPOCHS=9999 SENPAI_TIMEOUT_MINUTES=180 python train.py --dataset drivaerml --optimizer adamw --lr 3e-4 --cosine-t-max 30 --no-use-ema --enable-fourier --model-layers 4 --model-hidden-dim 512 --model-heads 8 --epochs 999 --batch-size 1 --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 --max-train-batches 394 --max-eval-batches 200 --agent ray --wandb-name "ray/drivaerml-4L512d-lr3e4" --wandb-group "drivaerml-4L512d-lr-sweep"
```

**W&B run:** u0xxaklo
**Peak memory:** Single GPU (H200 97GB)

## What happened

**Hypothesis falsified.** Lower LR (3e-4 vs 5e-4) did NOT improve DrivAerML at 4L/512d, unlike TandemFoil where 2e-4 beat 3e-4 by 14.7%.

The key difference is likely dataset-specific: DrivAerML's 394-batch epochs with T_max=30 cosine cycling already provide sufficient optimization stability at lr=5e-4. The higher LR explores more aggressively at cosine peaks, reaching deeper basins. At lr=3e-4, the cosine peaks are less exploratory, and the model settles into a shallower minimum earlier (epoch 215 vs 256).

The fact that both runs completed exactly 267 epochs (same throughput) rules out any epoch-count explanation — lr=5e-4 simply finds a better solution given the same compute budget.

## Suggested follow-ups

1. **lr=7e-4 or lr=1e-3** — if 5e-4 > 3e-4, the optimal LR for DrivAerML 4L/512d may be higher, not lower. The wider model can absorb more aggressive updates.
2. **Grad-clip at lr=5e-4** — grad-clip was transformative for AirfRANS (46% improvement). At DrivAerML's current lr=5e-4, clipping could stabilize the cosine peak oscillations without sacrificing exploration.
3. **WD=1e-2 at lr=5e-4** — weight decay regularization helped AirfRANS; could prevent the late-epoch overfitting seen at epoch 267 (5.93% vs best 4.62%).

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2797 because ADVISOR: Closing as falsified hypothesis — clean negative result. W&B confirms val=4.937% vs 4.619% baseline, a 6.9% regression. Lower LR (3e-4) converges to a shallower minimum within the 180-min budget for DrivAerML at 4L/512d. This is consistent with the independent falsification in #2779 (4L/320d, 12.8% regression), confirming that DrivAerML needs at least lr=5e-4 at these architectures. No further lower-LR experiments warranted for DrivAerML. New experiments will explore gc=0.5 at lr=5e-4, the promising follow-up direction from the AirfRANS breakthrough..
